pkg/plugins/trivy/plugin.go

```diff
@@ -154,7 +155,15 @@ func (p *plugin) ParseReportData(ctx trivyoperator.PluginContext, imageRef strin
 		return vulnReport, secretReport, nil, err
 	}
 
-	registry, artifact, err := p.parseImageRef(imageRef, reports.Metadata.ImageID)
+	// Note: every Docker image is associated with 2 different SHAs:
+	// - image digest (this is what K8s shows in imageID field in kubectl get pod -o yaml) - and, for example,
+	// kube-prometheus-stack reports this value in kube_pod_container_info metric
+	// - image ID (this is what is visible in 'docker image ls command')
+	// Execute docker image ls --digests --no-trunc <image> to see both digest and ID.
+	// See https://stackoverflow.com/questions/56364643/whats-the-difference-between-a-docker-images-image-id-and-its-digest
+	var imageDigest = strings.Split(reports.Metadata.RepoDigests[0], "@")[1]
```

Comment:
@d-mankowski-synerise any reason to strip the `sha256@` prefix?

Reply:
copying from the [issue](https://github.com/aquasecurity/trivy-operator/issues/2259) you raised:

```
    "RepoDigests": [
      "<registry>/<image repo>@sha256:cf3e3b3a98edde46eac58d4745a2467b9da2bc49a11191565a893ffac38034a3"
    ],
```
so it is not stripping `sha256` prefix.

Reply:
ohh, right! mixed up the delimiters 👍🏼